src/server/http/HTTPResponder.cpp

```diff
@@ -37,6 +41,17 @@ void HTTPResponder::handleError(HTTPClient &client, config::ServerBlock *server,
 }
 
 void HTTPResponder::handleError(HTTPClient &client, config::ServerBlock *server,  config::RouteBlock *route, int code, bool doErrorPage) {
+	// Request authentication
+	if (code == 401) {
+		client.data.response.setResponse(
+			ResponseBuilder("HTTP/1.1")
+			.setStatus(401)
+			.setHeader("WWW-AUTHENTICATE", "Basic realm=\"Not-Apache\"")
+			.setHeader("Connection", "Close")
+			.build()
+		);
+		return ;
+	}
```

Comment:
check route block along with 401 error code if the header has to be set.

also dont set an entire response, only set the header